docs/spec.md

```diff
@@ -170,6 +173,82 @@ Informs the client that channels are no longer available.
 }
 ```
 
+### Client Advertise
+
+- Informs the server about available client channels.
+
+#### Fields
+
+- `op`: string `"client_advertise"`
+- `channels`: array of:
+  - `id`: number chosen by the client. The client may not reuse ids across multiple advertised channels.
+  - `topic`: string
+  - `encoding`: string `"json"`
+  - `schemaName`: string
+
+#### Example
+
+```json
+{
+  "op": "client_advertise",
+  "channels": [
+    {
+      "id": 1,
+      "topic": "foo",
+      "encoding": "json",
```

Comment:
If we wanted to expand to binary publishing in the future, is this sufficient? Would we need to distinguish between a json websocket message and a binary websocket message that contains json-encoded data?

Reply:
I think in the future we can also allow the client to send message data in binary form. This could e.g. look similar to the binary messages send by the server: 

| Bytes           | Type    | Description                     |
| --------------- | ------- | ------------------------------- |
| 1               | opcode  | 0x01                            |
| 4               | uint32  | channel id                     |
| remaining bytes | uint8[] | message payload                 |

The first byte of the incoming message would tell the server if it's an operation message (JSON encoded) or if it is a binary message. If it is a binary message, the channel `encoding` would tell the server how to interpret the message payload. So if it's `json`, the server would parse the payload as JSON.